src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs

```diff
@@ -233,5 +233,61 @@ private void AddProjectChanges(XDocument project, string childPath = null)
                 project.Root.Add(target);
             }
         }
+
+        [Fact]
+        public void It_copies_content_transitively()
+        {
+            var targetFramework = "net5.0";
+            var testProjectA = new TestProject()
+            {
+                Name = "A",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true
+            };
+            var testAssetA = _testAssetsManager.CreateTestProject(testProjectA).WithProjectChanges((path, p) =>
+            {
+                var ns = p.Root.Name.Namespace;
+                p.Root.Add(new XElement(ns + "ItemGroup",
+                    new XElement(ns + "Content", new XAttribute("Include", "a.txt"), new XAttribute("CopyToOutputDirectory", "PreserveNewest"))));
+            });
+            File.WriteAllText(Path.Combine(testAssetA.Path, testProjectA.Name, "a.txt"), "A");
+
+            var testProjectB = new TestProject()
+            {
+                Name = "B",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true
+            };
+            testProjectB.AdditionalItems.Add("ProjectReference", Path.Combine(testAssetA.Path, testProjectA.Name, $"{testProjectA.Name}.csproj"));
+            var testAssetB = _testAssetsManager.CreateTestProject(testProjectB);
+
+            var testProjectC = new TestProject()
+            {
+                Name = "C",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true
+            };
+            testProjectC.AdditionalProperties.Add("DisableTransitiveProjectReferences", "true");
+            testProjectC.AdditionalItems.Add("ProjectReference", Path.Combine(testAssetB.Path, testProjectB.Name, $"{testProjectB.Name}.csproj"));
```

Comment:
You can create multiple `TestProjects`, add them to the `ReferencedProjects` of others as appropriate, and then create them all at once with one `CreateTestProject` call.

Reply:
That prevents us from being able to add project changes when creating the asset, right? We need to be able to do that here to add the content with the right metadata.

Reply:
It makes it a bit more complicated.  There's an overload of `WithProjectChanges` that takes an `Action<string, XDocument>`.  The string you get passed will be the path to the project file, so you can write conditional logic to apply the right changes to the right file.  At least, that's how I think it's supposed to work.